src/components/cylc/Task.vue

```diff
@@ -39,6 +39,16 @@
           stroke-dasharray="157"
           v-bind:style="progressStyle"
         ></circle>
+        <!-- dot in the middle (small hub)
+               * position in the middle
+               * radius can be changed independently
+               * we can't just change the radius with CSS (SVG2 only).
```

Comment:
**[svg rant]**

It's so tempting to do something like this:

```css
#hub {
    radius: 0;
}
&.submitted {
    #hub {
        radius: 8;
        transition: radius 0.5s;
    }
}
&.submitting {
    #hub {
        radius: 15;
        transition: radius 0.5s;
    }
}

```

But alas not. CSS3/SVG1.1 have limited coupling. Sadly the horrendusly delayed SVG2 is still in [draft status](https://svgwg.org/svg2-draft/)!

Doing something as simple as changing the radius of a circle requires JS. CSS3 animations are amazing so it's annoying but sadly we can't use them to animate the task status icons.

**[/svg rant]**

Reply:
If using JS... I haven't had any case to use it, but Vue has a small API for animation/transitions: https://vuejs.org/v2/guide/transitions.html

In theory we could add hooks to the component lifecycle & listen on DOM operations (element added, or page loaded, or component created, etc), and instruct Vue to change statuses.

Reply:
a css pre processor such as stylus could also be used http://stylus-lang.com/docs/conditionals.html